src/Umbraco.Web/Editors/ContentController.cs

```diff
@@ -1763,85 +1766,89 @@ public DomainSave PostSaveLanguageAndDomains(DomainSave model)
                     throw new HttpResponseException(response);
                 }
             }
-            else
-            {
-                var wildcard = domains.FirstOrDefault(d => d.IsWildcard);
-                if (wildcard != null)
-                {
-                    Services.DomainService.Delete(wildcard);
-                }
-            }
 
-            // process domains
-            // delete every (non-wildcard) domain, that exists in the DB yet is not in the model
-            foreach (var domain in domains.Where(d => d.IsWildcard == false && model.Domains.All(m => m.Name.InvariantEquals(d.DomainName) == false)))
+            // Delete every domain that's in the database, but not in the model
+            foreach (var assignedDomain in assignedDomains.Where(d => (d.IsWildcard && language == null) || (d.IsWildcard == false && model.Domains.All(m => m.Name.InvariantEquals(d.DomainName) == false))))
             {
-                Services.DomainService.Delete(domain);
+                Services.DomainService.Delete(assignedDomain);
             }
```

Comment:
This now removes all domains not required by the model, so including the wildcard domain.